File: Boats to Save People.js
var numRescueBoats = function(people, limit) {
    people.sort((a,b) => b-a);
    var r = 0
    var l = people.length-1;
    var boat = 0;
    
    while(r <= l){
        if(people[r] + people[l] <= limit) l--;
        r++;
        boat++;
    }
    return boat;
};

File: Boats to Save People.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        count_boats = 0
        left, right = 0, len(people) - 1
        people.sort()
        while left <= right:
            weight_sum = people[left] + people[right] 
            if weight_sum <= limit:
                left += 1
            right -= 1
            count_boats += 1
        
        return count_boats
